src/Equinor.ProCoSys.IPO.Command/InvitationCommands/FillRfocStatuses/FillRfocStatusesCommandHandler.cs

```diff
@@ -0,0 +1,155 @@
+                }
+                _logger.LogInformation($"McPkgs updated with RfocAccepted: {mcPkgsUpdatedCount}");
+                _logger.LogInformation($"CommPkgs updated with RfocAccepted: {commPkgsUpdatedCount}");
+                _logger.LogInformation($"Invitations set to ScopeHandedOver: {invitationsSetToScopeHandedOverCount}");
+            }
+
+            return new SuccessResult<Unit>(Unit.Value);
+        }
+
+        private async Task<int> HandleMcPkgsAsync(List<Invitation> invitations, Project project, int count)
+        {
+            var mcPkgsInProject = invitations.Where(i => i.Type == DisciplineType.DP)
+                .SelectMany(i => i.McPkgs)
+                .ToList();
+
+            var mcPkgNosInProject = mcPkgsInProject.Select(m => m.McPkgNo).Distinct().ToList();
+
+            var pcsMcPkgs = await _mcPkgApiService.GetMcPkgsByMcPkgNosAsync(project.Plant, project.Name, mcPkgNosInProject);
+
+            foreach (var pcsMcPkg in pcsMcPkgs)
+            {
+                if (pcsMcPkg.OperationHandoverStatus == "ACCEPTED")
+                {
+                    var mcPkgsToUpdate = mcPkgsInProject.Where(m => m.McPkgNo == pcsMcPkg.McPkgNo).ToList();
+                    foreach (var mcPkg in mcPkgsToUpdate)
+                    {
+                        mcPkg.RfocAccepted = true;
+                        count++;
+                    }
+                }
+            }
+
+            return count;
+        }
+
+        private async Task<int> HandleCommPkgsAsync(List<Invitation> invitations, Project project, int count)
+        {
+            var commPkgsInProject = invitations.Where(i => i.Type == DisciplineType.MDP)
+                .SelectMany(i => i.CommPkgs)
+                .ToList();
+
+            var commPkgNosInProject = commPkgsInProject.Select(c => c.CommPkgNo).Distinct().ToList();
+
+            var pcsCommPkgs = await _commPkgApiService.GetCommPkgsByCommPkgNosAsync(project.Plant, project.Name, commPkgNosInProject);
+
+            foreach (var pcsCommPkg in pcsCommPkgs)
+            {
+                if (pcsCommPkg.OperationHandoverStatus == "ACCEPTED")
+                {
+                    var commPkgsToUpdate = commPkgsInProject.Where(m => m.CommPkgNo == pcsCommPkg.CommPkgNo).ToList();
+                    foreach (var commPkg in commPkgsToUpdate)
+                    {
+                        commPkg.RfocAccepted = true;
+                        count++;
+                    }
+                }
+            }
+
+            return count;
+        }
+
+        private static int HandleInvitationStatus(List<Invitation> invitations, int count)
+        {
+            foreach (var invitation in invitations)
+            {
+                if (invitation.Type == DisciplineType.DP)
+                {
+                    if (invitation.McPkgs.All(m => m.RfocAccepted))
+                    {
+                        invitation.ScopeHandedOver();
```

Comment:
.ScopeHandedOver() is allowed just for invitations with certain statuses, and will cause exception if not allowed. I can't see that you select only such invitations  (maybe the Fill-operation should be allowed for all statuses?)

Reply:
I changed to not get IPOs with status Canceled or ScopeHandedOver.
Made a change to update IPOs with status Accepted to ScopeHandedOver as well, as this is neccessary to remove IPOs from other participants TODO-list who have not signed. History still shows if IPO was accepted.